bootstrap.py

```diff
@@ -354,6 +368,57 @@ def apply_configuration(node_set, configuration_file, global_patches):
     # Add Helm repo for Cilium
     helm("repo", "add", "cilium", "https://helm.cilium.io/")
 
+    # Discover resource types from the API server
+    resource_types = []
+    namespaced_count = 0
+    for namespaced in ["true", "false"]:
+        r = kubectl(
+            "api-resources",
+            "--verbs=list",
+            "-o=name",
+            f"--namespaced={namespaced}",
+            capture_stdout=True,
+        )
+        resource_types += [
+            t for t in r.stdout.strip().split('\n')
+            if t not in {
+                # Exclude resource types that cause trouble
+                "componentstatuses",
+                "validatingadmissionpolicies.admissionregistration.k8s.io",
+                "validatingadmissionpolicybindings.admissionregistration.k8s.io",
+            }
+        ]
+        namespaced_count = namespaced_count or len(resource_types)
+
+    print("> Cleaning up Flannel resources...")
+    for i, resource_type in enumerate(resource_types):
+        # Talos resource declarations: `pkg/flannel/template.go`
+        r = kubectl(
+            "--namespace=kube-system" if i < namespaced_count else None,
+            "delete",
+            "--ignore-not-found",
+            "--selector=k8s-app=flannel",
+            resource_type,
+            capture_stdout=True,
+            silent=True,
+        )
+        print(
+            '\n'.join((l for l in r.stdout.split('\n') if l != "No resources found")),
```

Comment:
Rename `l` to `line` perhaps?

Reply:
Yeah, refactored to avoid single character variable names entirely.